test/jdk/javax/swing/plaf/basic/BasicTextUI/8001470/bug8001470.java

```diff
@@ -26,45 +26,76 @@
  * @key headful
  * @bug 8001470
  * @summary JTextField's size is computed incorrectly when it contains Indic or Thai characters
- * @author Semyon Sadetsky
  */
 
-import javax.swing.*;
-import java.awt.*;
+import java.awt.Font;
+import java.awt.GraphicsEnvironment;
+import java.awt.GridLayout;
+import javax.swing.JFrame;
+import javax.swing.JTextField;
+import javax.swing.SwingUtilities;
+import javax.swing.JPanel;
 
 public class bug8001470 {
 
     private static JFrame frame;
     private static JTextField textField1;
     private static JTextField textField2;
+    private static boolean fontFound = false;
```

Comment:
```suggestion
    private static volatile boolean fontFound = false;
```
The value is set on EDT, but the value is read on the main thread.